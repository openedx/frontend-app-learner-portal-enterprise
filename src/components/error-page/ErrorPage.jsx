import React from 'react';
import PropTypes from 'prop-types';
import {
  Col,
} from '@openedx/paragon';

import SiteFooter from '@edx/frontend-component-footer';
import ErrorPageHeader from './ErrorPageHeader';
import ErrorPageTitle from './ErrorPageTitle';
import ErrorPageSubtitle from './ErrorPageSubtitle';
import ErrorPageContent from './ErrorPageContent';

/**
 * React component for the error case when attempting to link a user to a customer. Renders
 * a header, error alert, and a footer.
 */

const ErrorPage = ({
  title,
  subtitle,
  showSiteHeader,
  showSiteFooter,
  children,
}) => (
  <>
    {showSiteHeader && <ErrorPageHeader />}
    <main id="content">
      <ErrorPageContent>
        <Col xs={12} lg={{ span: 10, offset: 1 }}>
          <ErrorPageTitle>{title}</ErrorPageTitle>
          {subtitle && (
            <ErrorPageSubtitle>{subtitle}</ErrorPageSubtitle>
          )}
          {children}
        </Col>
      </ErrorPageContent>
    </main>
    {showSiteFooter && <SiteFooter />}
  </>
);

ErrorPage.Content = ErrorPageContent;
ErrorPage.Title = ErrorPageTitle;
ErrorPage.Subtitle = ErrorPageSubtitle;

ErrorPage.propTypes = {
  showSiteHeader: PropTypes.bool,
  children: PropTypes.node.isRequired,
  title: PropTypes.node,
  subtitle: PropTypes.node,
  showSiteFooter: PropTypes.bool,
};

ErrorPage.defaultProps = {
  title: 'Error occurred while processing your request',
  subtitle: null,
  showSiteHeader: true,
  showSiteFooter: true,
};

export default ErrorPage;
