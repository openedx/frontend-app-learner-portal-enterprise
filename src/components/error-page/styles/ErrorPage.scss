.error-page-container {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.error-page-title {
  color: $danger;
  margin-bottom: map_get($spacers, 5);
}

.error-page-subtitle {
  margin-bottom: map_get($spacers, 4);
}
