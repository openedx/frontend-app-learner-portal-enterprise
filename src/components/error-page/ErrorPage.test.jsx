import React from 'react';
import renderer from 'react-test-renderer';
import { AppContext } from '@edx/frontend-platform/react';
import '@testing-library/jest-dom/extend-expect';

import ErrorPage from './ErrorPage';

jest.mock('@edx/frontend-component-footer', () => function MockFrontendComponentFooter() {
  return <div data-testid="site-footer" />;
});

const ErrorPageWrapper = (component) => renderer
  .create((
    <AppContext.Provider value={{
      authenticatedUser: {
        id: 1,
        profileImage: {
          imageUrlMedium: 'htts://img.url',
        },
      },
      config: {
        LEARNER_SUPPORT_URL: 'https://support.url',
        LOGOUT_URL: 'https://logout.url',
      },
    }}
    >
      {component}
    </AppContext.Provider>
  ));

describe('ErrorPage', () => {
  test('properly renders page layout with title and subtitle', async () => {
    const tree = ErrorPageWrapper(
      <ErrorPage
        title="Something went wrong"
        subtitle="More details here"
      >
        Here goes the error message.
      </ErrorPage>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handles optional title', async () => {
    const tree = ErrorPageWrapper(
      <ErrorPage>
        Here goes the error message.
      </ErrorPage>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handles optional subtitle', async () => {
    const tree = ErrorPageWrapper(
      <ErrorPage>
        Here goes the error message.
      </ErrorPage>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
