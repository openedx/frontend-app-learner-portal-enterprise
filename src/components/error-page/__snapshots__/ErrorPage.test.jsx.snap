// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage handles optional title and subtitle 1`] = `
[
  <header>
    <nav
      className="border-bottom navbar navbar-expand-lg navbar-light bg-white"
      role="navigation"
    >
      <div
        className="container-fluid"
      >
        <nav
          className="navbar"
        >
          <img
            alt="edX logo"
            src="icon/mock/path"
            width={50}
          />
        </nav>
        <nav
          aria-label="Secondary"
          className="nav secondary-menu-container align-items-center ml-auto"
        >
          <a
            className="text-gray-700 mr-3"
            href="https://support.url"
          >
            Help
          </a>
        </nav>
      </div>
    </nav>
  </header>,
  <main
    className="fill-vertical-space"
    id="content"
  >
    <div
      className="error-page-container container-mw-lg container-fluid"
    >
      <div
        className="col-lg-10 col-12 offset-lg-1"
      >
        <h2
          className="text-danger mb-4"
        >
          Error occurred while processing your request
        </h2>
        Here goes the error message.
      </div>
    </div>
  </main>,
  <div
    data-testid="site-footer"
  />,
]
`;

exports[`ErrorPage properly renders page layout with title and subtitle 1`] = `
[
  <header>
    <nav
      className="border-bottom navbar navbar-expand-lg navbar-light bg-white"
      role="navigation"
    >
      <div
        className="container-fluid"
      >
        <nav
          className="navbar"
        >
          <img
            alt="edX logo"
            src="icon/mock/path"
            width={50}
          />
        </nav>
        <nav
          aria-label="Secondary"
          className="nav secondary-menu-container align-items-center ml-auto"
        >
          <a
            className="text-gray-700 mr-3"
            href="https://support.url"
          >
            Help
          </a>
        </nav>
      </div>
    </nav>
  </header>,
  <main
    className="fill-vertical-space"
    id="content"
  >
    <div
      className="error-page-container container-mw-lg container-fluid"
    >
      <div
        className="col-lg-10 col-12 offset-lg-1"
      >
        <h2
          className="text-danger mb-4"
        >
          Something went wrong
        </h2>
        <h3
          className="mb-4.5 text-gray-700"
        >
          More details here
        </h3>
        Here goes the error message.
      </div>
    </div>
  </main>,
  <div
    data-testid="site-footer"
  />,
]
`;
